syntax = "proto3";

package mediocre.video.v1beta;

import "mediocre/configuration/v1beta/game.proto";

service VideoService {
  rpc Video(VideoRequest) returns (stream VideoResponse);
}

message VideoRequest {
  configuration.v1beta.GameConfiguration configuration = 1;
  string source = 2;
}

message VideoResponse {
  double timestamp = 1;
  repeated RegionResponse regions = 2;
}

message RegionResponse {
  string name = 4;
  string output = 5;
}
